packages/fuel-indexer-lib/src/helpers.rs

```diff
@@ -0,0 +1,47 @@
+use fuel_abi_types::abi::program::TypeDeclaration;
+
+/// Whether a `TypeDeclaration` is tuple type
+pub fn is_tuple_type(typ: &TypeDeclaration) -> bool {
+    let mut type_field_chars = typ.type_field.chars();
+    type_field_chars.next().is_some_and(|c| c == '(')
+        && type_field_chars.next().is_some_and(|c| c != ')')
+}
+
+/// Whether a `TypeDeclaration` is a unit type
+pub fn is_unit_type(typ: &TypeDeclaration) -> bool {
```

Comment:
@lostman 
- Ah ok, yea, as I mentioned in the "Is there any particular reason..." comment...
  - _All_ of this codegen logic needs to be moved to `fuel-indexer-macros` (we're duplicating a ton of logic here)
  
So maybe something like:
- `fuel_indexer_macros::schema_gen`
  - This is where you put all the logic related to auto-creating this schema
    - And you would just pull in `fuel_indexer_macros::helpers` instead of duplicating this stuff
    - We already have a `fuel_indexer_macro_utils` crate
       - This was previously being used for indexer-internal proc macros (as opposed to public macros)
       - You could potentially move that `#[metrics]` macro out of `fuel_indexer_macro_utils` and into `fuel_indexer_macros` (so now fuel_indexer_macros would export `#[metrics]` and `#[indexer`]
       - _Then_ that would allow you to put your macro logic in `fuel_indexer_macro_utils` (after you make it a non-proc-macro crate)
       - Finally, this would allow you to import the macro functions into `forc_index::new` via `use fuel_indexer_macro_utils::schema_gen::*`
         - I think this solves a lot of problems but let me know what you think

Reply:
@lostman I mentioned in my previous comment

> Then that would allow you to put your macro logic in fuel_indexer_macro_utils (after you make it a non-proc-macro crate)